pkgs/development/python-modules/hypercorn/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitLab
+, pythonOlder
+, typing-extensions
+, wsproto
+, toml
+, h2
+, priority
+, pytest
+, pytest-asyncio
+, pytest-cov
+, pytest-sugar
+, pytest-trio
+}:
+
+buildPythonPackage rec {
+  pname = "Hypercorn";
+  version = "0.11.2";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitLab {
+    owner = "pgjones";
+    repo = pname;
+    rev = version;
+    sha256 = "0v80v6l2xqac5mgrmh2im7y23wpvz4yc2v4h9ryhvl88c2jk9mvh";
+  };
+
+  propagatedBuildInputs = [ wsproto toml h2 priority ]
+  ++ lib.optionals (pythonOlder "3.8") [ typing-extensions ];
+
+  checkInputs = [
+    pytest
+    pytest-asyncio
+    pytest-cov
+    pytest-sugar
+    pytest-trio
+    ];
+
+  checkPhase = "pytest";
```

Comment:
Would you please use `pytestCheckHook` instead? This is a bit more nix friendly, and also ensures that tests are run as part of a build.

More information can be found on the [pytestCheckHook](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook) section of the manual.

Reply:
Done, thank you.